tests/test_fastavro.py

```diff
@@ -595,6 +596,119 @@ def test_schema_migration_reader_union():
     assert new_records == records
 
 
+def test_aliases_in_reader_schema():
+    schema = {
+        "type": "record",
+        "fields": [{
+            "name": "test",
+            "type": "int"
+        }]
+    }
+
+    new_schema = {
+        "type": "record",
+        "fields": [{
+            "name": "newtest",
+            "type": "int",
+            "aliases": ["test"]
+        }]
+    }
+
+    new_file = MemoryIO()
+    records = [{"test": 1}]
+    fastavro.writer(new_file, schema, records)
+    new_file.seek(0)
+    new_reader = fastavro.reader(new_file, new_schema)
+    new_records = list(new_reader)
+    assert new_records[0]["newtest"] == records[0]["test"]
+
+
+def test_unused_aliases_in_reader_schema():
+    schema = {
+        "type": "record",
+        "fields": [
+            {
+                "name": "test1",
+                "type": "int"
+            },
+            {
+                "name": "test2",
+                "type": "int"
+            }
+        ]
+    }
+
+    new_schema = {
+        "type": "record",
+        "fields": [{
+            "name": "test2",
+            "type": "int",
+            "aliases": ["test1"]
+        }]
+    }
+
+    new_file = MemoryIO()
+    records = [{"test1": 1, "test2": 2}]
+    fastavro.writer(new_file, schema, records)
+    new_file.seek(0)
+    new_reader = fastavro.reader(new_file, new_schema)
+    new_records = list(new_reader)
+    assert new_records[0]["test2"] == records[0]["test2"]
+
+
+def test_same_alias_for_multiple_fields():
```

Comment:
I think this test case and feature is the reason for the `rewindable_reader`. I'm curious to know if this is a real scenario you experience right now. It seems strange to me that two fields on a new schema would alias the same old field...

Reply:
Yes, exactly. We would not need the (admittedly hackish) `rewindable_reader` if we did not have to validate data against multiple types.
You could decide to support aliases, but not multiple aliases to the same name. The current implementation would be very much simplified. Having multiple fields point to the same aliased field has never occurred to me, but I can imagine how it could happen: when a field becomes superseded by another, it is still used downstream for some time. It might be important to provide correct values for this legacy field, at least for some time.

Reply:
If you don't have a need for it then I think it's best to wait until there is a need. It can be implemented at that time and until then it will make things simpler.

Reply:
I'm fine with that.

Reply:
I can confirm that the reference Java implementation does *not* support multiple fields with the same alias. I'm getting a `org.apache.avro.AvroTypeException` whenever I try to read data with multiple fields that share a common alias. This is good news (sorta) because it should allow me to really simplify the code.